src/MUST_analysis_tools/MUST_CP_long_analysis.py

```diff
@@ -0,0 +1,230 @@
+        confidence_stdev_veh = np.std(confidences)
+        confidence_pct_below_limit_veh = 100 * len(confidences[confidences > 90]) / len(confidences)
+        confidence_mean += confidence_mean_veh
+        confidence_stdev += confidence_stdev_veh
+        if confidence_pct_below_limit_veh > 90:
+            passed_count += 1
+    confidence_mean /= len(vehicle_ids)
+    confidence_stdev /= len(vehicle_ids)
+    passed_pct_confidence = 100 * passed_count / len(vehicle_ids)
+    print(f'Metric 6, confidence score >90% per-vehicle data. Mean confidence: {confidence_mean}, stdev: {confidence_stdev}, percentage >90%: {passed_pct_confidence}%')
+    metric_6_pass = bool(passed_pct_confidence >= 90)
+
+    # Metric 7: detection frequency (30hz +- 3hz)
+    deduplicated_timestamp_list = np.sort(np.unique(must_data['epoch_time'].to_numpy()))
+    recording_periods = np.diff(deduplicated_timestamp_list)
+    freq_mean = 1 / np.mean(recording_periods)
+    freq_stdev = np.std(1 / recording_periods)
+    freq_pct_below_limit = 100 * len(recording_periods[recording_periods < (1.0 / (30 - 3))]) / len(recording_periods)
+    # insert histogram. X axis frequency (0-30), Y axis percentage
+    print(f'Metric 7, detection frequency 30hz +- 3hz. Mean frequency: {freq_mean:.2f}, stdev: {freq_stdev:.2f}, percentage above 27hz: {freq_pct_below_limit:.2f}%')
+    metric_7_pass = bool(freq_pct_below_limit >= 90)
+
+    # Metric 8: 90% of vehicle IDs do not fluctuate
+    vehicle_ids = np.unique(must_data['vehicle id'].to_numpy())
+    track_counts = []
+    for vehicle_id in vehicle_ids:
+        this_vehicle_data = must_data[must_data['vehicle id'] == vehicle_id]
+        num_tracks = len(find_sub_tracks(this_vehicle_data['epoch_time'].to_numpy(),
+                                        this_vehicle_data['latitude'].to_numpy(), this_vehicle_data['longitude'].to_numpy())) - 2
+        track_counts.append(num_tracks)
+    track_counts = np.array(track_counts)
+    tracks_mean = np.mean(track_counts)
+    tracks_stdev = np.std(track_counts)
+    tracks_pct_below_limit = 100 * len(track_counts[track_counts < 1]) / len(track_counts)
+    print(f'Metric 8, >90% of IDs do not fluctuate. Mean number of ID swaps: {tracks_mean:.2f}, stdev: {tracks_stdev:.2f}, percentage that did not fluctuate: {tracks_pct_below_limit}%')
+    metric_8_pass = bool(tracks_pct_below_limit >= 90)
+
+    with open(os.path.join(output_folder, f'long_metrics.csv'), 'a') as outfile:
+        writer = csv.writer(outfile)
+        writer.writerow([test_name, confidence_mean, confidence_stdev, passed_pct_confidence, int(metric_6_pass),
+                         freq_mean, freq_stdev, freq_pct_below_limit, int(metric_7_pass),
+                         tracks_mean, tracks_stdev, tracks_pct_below_limit, int(metric_8_pass)])
+
+
+def main(args):
+    ## Folder/data paths
+    if len(args) != 5:
+        print('Please use the format: python3 MUST_CP_long_analysis.py '
```

Comment:
Using python [argsparse ](https://docs.python.org/3/library/argparse.html) would allow you to provide a bit more details about each file and folder arguments?

Reply:
Is argparse helpful when it's all required arguments?

Reply:
At the very least it allows you to add help text that will print out when arguments are missing or of an incorrect type. Seems relatively low effort  to introduce unless I am missing some limitation. Also seems more robust then simply checking the length of arguments provided.